Panic: check backup without snapshots

I just created a new restic repository but haven't created any snapshot, yet. The check invocation failed:

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.selectRandomPacksByPercentage(0xc00007c960, 0x3fc3333333333333, 0x3fc3333333333333)
        /restic/cmd/restic/cmd_check.go:349 +0x4d0                                      
main.runCheck(0xc0003a9900, 0x7fff92507ef8, 0x5, 0xc0003b0000, 0x7fff92507ec6, 0x15, 0x0, 0x0, 0x0, 0x0, ...)
        /restic/cmd/restic/cmd_check.go:299 +0x1206
main.glob..func5(0x16590c0, 0xc00038ea40, 0x0, 0x4, 0x0, 0x0)
        /restic/cmd/restic/cmd_check.go:35 +0xcc
github.com/spf13/cobra.(*Command).execute(0x16590c0, 0xc0003b67e0, 0x4, 0x6, 0x16590c0, 0xc0003b67e0)
        /home/build/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x47c                   
github.com/spf13/cobra.(*Command).ExecuteC(0x165bb40, 0xc00037d6c0, 0x2, 0x8)
        /home/build/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
        /home/build/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        /restic/cmd/restic/main.go:98 +0x3f
```

This is with restic 0.12.0 on Linux amd64.

--- a/changelog/unreleased/issue-3296
+++ b/changelog/unreleased/issue-3296
@@ -1,0 +1,7 @@
+Bugfix: Fix crash of `check --read-data-subset=x%` run for an empty repository
+
+`check --read-data-subset=x%` crashed when run for an empty repository. This
+has been fixed.
+
+https://github.com/restic/restic/issues/3296
+https://github.com/restic/restic/pull/3309

--- a/cmd/restic/cmd_check.go
+++ b/cmd/restic/cmd_check.go
@@ -331,7 +331,7 @@ func selectPacksByBucket(allPacks map[restic.ID]int64, bucket, totalBuckets uint
 func selectRandomPacksByPercentage(allPacks map[restic.ID]int64, percentage float64) map[restic.ID]int64 {
 	packCount := len(allPacks)
 	packsToCheck := int(float64(packCount) * (percentage / 100.0))
-	if packsToCheck < 1 {
+	if packCount > 0 && packsToCheck < 1 {
 		packsToCheck = 1
 	}
 	timeNs := time.Now().UnixNano()

--- a/cmd/restic/cmd_check_test.go
+++ b/cmd/restic/cmd_check_test.go
@@ -122,3 +122,10 @@ func TestSelectRandomPacksByPercentage(t *testing.T) {
 		rtest.Assert(t, ok, "Expected input and output to be equal")
 	}
 }
+
+func TestSelectNoRandomPacksByPercentage(t *testing.T) {
+	// that the a repository without pack files works
+	var testPacks = make(map[restic.ID]int64)
+	selectedPacks := selectRandomPacksByPercentage(testPacks, 10.0)
+	rtest.Assert(t, len(selectedPacks) == 0, "Expected 0 selected packs")
+}

